### DISCUSSION TITLE
Is Jedis 2.5.2 fully compatible with redis 2.8.x, 3.x.x and above*?

### DISCUSSION THREAD

**Discussion by jiandanfeng** [xetorthio/jedis#2405] [Q&A]
Is Jedis 2.5.2 fully compatible with redis 2.8.x, 3.x.x and above*?

**Comment by gkorland** [comment]
Jedis 3.0.0 has some breaking changes

**Comment by vadimPT** [comment]
I am running Redis server v=3.2.100. Trying to subscribe to a channel with the latest Jedis freezes the system although method onSubscribe is called: no progress after calling jSubscriber.subscribe(jedisPubSub, "channel"), i.e. no instructions executed after this call and method onMessage is not called although messages are published into the appropriate channel.

Is it due to the breaking changes that you have mentioned? Do you know when the update will be available that solves such issues?

**Comment by gkorland** [comment]
No, that should work, can you please post a full recreation?

**Comment by vadimPT** [comment]
The structure is the following.

            jSubscriber = new Jedis("redis", 6379);
            JedisPubSub jedisPubSub = new JedisPubSub() {

                @Override
                public void onSubscribe(String channel, int subscribedChannels) {
                    System.out.println("Subscribed to channel " + channel);
                }

                @Override
                public void onMessage(String channel, String message) {
                    System.out.println("Message via channel " + channel);
                }
            };
            jSubscriber.subscribe(jedisPubSub, "CMD");
            doSomething();

onSubscribe is executed. Also if use jSubscriber after creation to write data into the Redis instance it goes well and I see the data inside the DB.

doSomething() is never called.

**Comment by MichalCho** [comment]
I think you need a separate thread for executing subscribe as the client has to react on coming messages. So you could try to add some code like 
```
Thread one = new Thread(() -> {
      try {
        jSubscriber.subscribe(jedisPubSub, "CMD");
      } catch (Exception e) {
        System.out.println("Subscribing failed."+ e);
      }
    });
one.start();
```

**Comment by chrisjenx** [comment]
> I think you need a separate thread for executing subscribe as the client has to react on coming messages. So you could try to add some code like
> 
> ```
> Thread one = new Thread(() -> {
>       try {
>         jSubscriber.subscribe(jedisPubSub, "CMD");
>       } catch (Exception e) {
>         System.out.println("Subscribing failed."+ e);
>       }
>     });
> one.start();
> ```

Yeah this isn't really clear, would be great to get this added to the jdoc that it blocks the current thread.

**Comment by rkimsb2** [comment]
Hello. I have same question on Redis 4.x.

Should I separate the question for this?

**Comment by sazzad16** [comment]
@rkimsb2 Sure! Please mention which incompatibility you are having. It would be hard to help with a generic question.
